File: JavaPop/app/src/main/java/com/example/javapop/Adapter/AdapterPullRequest.java
package com.example.javapop.Adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.javapop.Events.ClickItemPullRequestEvent;
import com.example.javapop.Models.PullRequest;
import com.example.javapop.Models.PullRequestList;
import com.example.javapop.R;
import com.example.javapop.Utils.CircleTransform;
import com.squareup.picasso.Picasso;

import org.greenrobot.eventbus.EventBus;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.List;
import java.util.TimeZone;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class AdapterPullRequest extends RecyclerView.Adapter<AdapterPullRequest.ViewHolderRepository> {

    private List<PullRequestList> pullRequests;
    private Context mContext;

    public static class ViewHolderRepository extends RecyclerView.ViewHolder {

        @BindView(R.id.title_PR)
        TextView mTitlePR;

        @BindView(R.id.description_PR)
        TextView mDescriptionPR;

        @BindView(R.id.image_user_PR)
        ImageView mImagePR;

        @BindView(R.id.name_username_PR)
        TextView mNameUsernamePR;

        @BindView(R.id.date_PR)
        TextView mDatePR;

        PullRequest mPullRequest;

        @OnClick(R.id.item_pull_request)
        void itemPullRequest() {
            EventBus.getDefault().post(new ClickItemPullRequestEvent(mPullRequest, getAdapterPosition()));
        }

        public ViewHolderRepository(View itemView) {
            super(itemView);
            ButterKnife.bind(this, itemView);
        }

        private void getPullRequest(PullRequest pullRequest) {
            this.mPullRequest = pullRequest;
        }
    }

    public AdapterPullRequest(List<PullRequestList> pullRequest, Context context) {
        this.pullRequests = pullRequest;
        this.mContext = context;
    }

    @NonNull
    @Override
    public ViewHolderRepository onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.layout_item_pull_request, parent, false);
        return new ViewHolderRepository(v);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolderRepository holder, int position) {
        PullRequest pullRequest = pullRequests.get(position);

        holder.mTitlePR.setText(pullRequest.getTitlePR());
        holder.mDescriptionPR.setText(pullRequest.getDescriptionPR());
        Picasso.with(mContext).load(pullRequest.getOwnerGit().getURLimage()).
                transform(new CircleTransform()).into(holder.mImagePR);
        holder.mNameUsernamePR.setText(pullRequest.getOwnerGit().getUsername());
        holder.mDatePR.setText(dateFormat(pullRequest.getDatePR()));

        holder.getPullRequest(pullRequest);
    }

    @Override
    public int getItemCount() {
        return pullRequests.size();
    }

    @SuppressLint("SimpleDateFormat")
    private String dateFormat(String stringDate) {
        try {
            Date date = null;
            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'");
            sdf.setTimeZone(TimeZone.getTimeZone("GMT"));
            date = sdf.parse(stringDate);
            return new SimpleDateFormat("dd/MM/yyyy").format(date);
        } catch (ParseException e) {
            e.printStackTrace();
        }
        return "";
    }
}

File: app/src/main/java/cl/carteaga/querygithub/classes/AdapterPullRequest.kt
package cl.carteaga.querygithub.classes

import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.view.View
import android.view.ViewGroup
import com.bumptech.glide.Glide
import kotlinx.android.synthetic.main.item_pull_request.view.*
import android.net.Uri
import cl.carteaga.querygithub.R
import cl.carteaga.querygithub.models.PullRequest
import com.bumptech.glide.load.engine.DiskCacheStrategy


class AdapterPullRequest(val data: List<PullRequest>?) :
    RecyclerView.Adapter<AdapterPullRequest.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return AdapterPullRequest.ViewHolder(parent.inflate(R.layout.item_pull_request))
    }

    override fun getItemCount(): Int = data?.size?: 0

    override fun onBindViewHolder(holder: AdapterPullRequest.ViewHolder, position: Int) {
        holder.bindView(data?.get(position))
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bindView(pullRequest: PullRequest?) {
            itemView.txtTitlePullRequest.text = pullRequest?.title
            itemView.txtBodyPullRequest.text = pullRequest?.body
            itemView.txtAuthorNamePullRequest.text = pullRequest?.user?.login
            Glide.with(itemView.context)
                .load(pullRequest?.user?.avatarUrl)
                .diskCacheStrategy(DiskCacheStrategy.ALL)
                .into(itemView.imgAvatarPullRequest)

            itemView.setOnClickListener {
                itemView.context
                    .startActivity(
                        Intent(
                            Intent.ACTION_VIEW,
                            Uri.parse(pullRequest?.htmlUrl
                            )
                        )
                    )
            }
        }
    }
}